src/metabase/models/user_parameter_value.clj

```diff
@@ -52,17 +56,19 @@
   "Hydrate a map of parameter-id->last-used-value for the dashboards."
   [_model _k dashboards]
   (if-let [user-id api/*current-user-id*]
-    (let [all-parameter-ids               (into #{} (comp (mapcat :parameters) (map :id)) dashboards)
-          parameter-ids->last-used-values (when (seq all-parameter-ids)
-                                            (into {}
-                                                  (t2/select-fn-vec
-                                                   (fn [{:keys [parameter_id value]}]
-                                                     [parameter_id value])
-                                                   :model/UserParameterValue
-                                                   :user_id user-id
-                                                   :parameter_id [:in all-parameter-ids])))]
+    (let [all-parameter-ids (into #{} (comp (mapcat :parameters) (map :id)) dashboards)
+          last-used-values  (fn [dashboard-id]
+                              (when (seq all-parameter-ids)
+                                (into {}
+                                      (t2/select-fn-vec
+                                       (fn [{:keys [parameter_id value]}]
+                                         [parameter_id value])
+                                       :model/UserParameterValue
+                                       :user_id user-id
+                                       :dashboard_id dashboard-id
+                                       :parameter_id [:in all-parameter-ids]))))]
```

Comment:
It is more performant to use this `[model & column]` form, and it should return the same thing:

```suggestion
                                      (t2/select
                                       [:model/UserParameterValue :parameter_id :value]
                                       :user_id user-id
                                       :dashboard_id dashboard-id
                                       :parameter_id [:in all-parameter-ids]))))]
```

Reply:
actually, the shape's not quite right so I think I'll keep it the way it was. 
Might look at alternatives to see if there's something more performant that still works the way I need.